tooling/src/main/java/org/opencds/cqf/tooling/casereporting/tes/TESPackageGenerator.java

```diff
@@ -0,0 +1,615 @@
+        inputParameters.version = version;
+        inputParameters.releaseLabel = releaseLabel;
+        inputParameters.pathToInputBundle = pathToInputBundle;
+        inputParameters.pathToConditionGrouperWorkbook = pathToConditionGrouperWorkbook;
+        inputParameters.pathToConditionCodeValueSet = pathToConditionCodeValueSet;
+        inputParameters.outputPath = this.getOutputPath();
+        inputParameters.outputFileName = outputFileName;
+        inputParameters.outputFileEncodings = outputFileEncodings;
+        inputParameters.writeConditionGroupers = writeConditionGroupers;
+
+        return generatePackage(inputParameters);
+    }
+
+    private void loadOperationArguments(TESPackageGenerateParameters params) {
+        version = params.version;
+        releaseLabel = params.releaseLabel;
+        pathToInputBundle = params.pathToInputBundle;
+        pathToConditionGrouperWorkbook = params.pathToConditionGrouperWorkbook;
+        pathToConditionCodeValueSet = params.pathToConditionCodeValueSet;
+        setOutputPath(params.outputPath);
+        outputFileName = params.outputFileName;
+        outputFileEncodings = params.outputFileEncodings;
+        writeConditionGroupers = params.writeConditionGroupers;
+    }
+
+    public Bundle generatePackage(TESPackageGenerateParameters params) {
+        loadOperationArguments(params);
+        List<ConditionGroupingEntry> conditionGroupingEntries = loadConditionGroupingDefinitions(pathToConditionGrouperWorkbook);
+        List<ValueSet> conditionGroupers = generateConditionGroupers(conditionGroupingEntries);
+        List<ValueSet> reportingSpecificationGroupers = loadReportingSpecificationGroupers(params.pathToInputBundle);
+        addReportingSpecificationGrouperReferencesToConditionGroupers(conditionGroupingEntries, conditionGroupers, reportingSpecificationGroupers);
+        Library manifest = generateManifest(conditionGroupers, reportingSpecificationGroupers);
+
+        // Write ConditionGroupers out to files (should be an argument)
+        if (writeConditionGroupers) {
+            for (IOUtils.Encoding encoding : getOutputFileEncodings()) {
+                IOUtils.writeResources(conditionGroupers, this.getOutputPath(), encoding, FhirContext.forR4Cached());
+            }
+        }
+
+        List<IBaseResource> resourcesToBundle = new ArrayList<>();
+        resourcesToBundle.add(manifest);
+        resourcesToBundle.addAll(conditionGroupers);
+        resourcesToBundle.addAll(reportingSpecificationGroupers);
+
+        Bundle outputBundle = null;
+        outputBundle = buildAndWriteBundle(resourcesToBundle);
+
+        if (!params.pathToConditionCodeValueSet.isEmpty()) {
+            try {
+                generateConditionCodeUsageComparison(params.pathToConditionCodeValueSet, reportingSpecificationGroupers);
+            } catch (Exception ex) {
+                throw new RuntimeException("Unable to create condition code difference workbook file.");
```

Comment:
Typically do not want a generic exceptions... Also losing the stack trace and message from the caught exception.